src/dune_lang/package_dependency.ml

```diff
@@ -40,12 +40,64 @@ let encode { name; constraint_ } =
   | Some c -> pair Package_name.encode Package_constraint.encode (name, c)
 ;;
 
+(* Check for common typos in package dependency constraints *)
+let check_for_typo ~loc { name; constraint_ } =
+  (* Only check for typos if warnings are enabled *)
+  if Dune_config.Config.(get typo_warnings) = `Enabled
+  then
+    let open Package_constraint in
```

Comment:
I'd probably avoid the local open here or if it is this verbose, then use `let module C = Package_constraint in` to avoid potentially pulling in unexpected values.